src/tools/compiletest/src/header.rs

```diff
@@ -1028,6 +1039,19 @@ fn iter_header(
 
                     return;
                 }
+
+                if let Some(trailing_directive) = &trailing_directive {
+                    *poisoned = true;
+
+                    eprintln!(
+                        "error: detected trailing compiletest test directive `{}` in {}:{}",
```

Comment:
At the cost of verbosity, should we also recommend what the user should do, i.e. some wording like "one directive per line"? Or maybe reword this as something like "each line can only contain one compiletest directive"?

Reply:
I've added a help line.

```
error: detected trailing compiletest test directive `only-arm` in a.rs:1
 help: put the trailing directive in it's own line: `//@ only-arm`
```

Reply:
Hmm, I wonder if we should say "each line can only contain one compiletest directive" instead of saying both trailing detected + the help, because if someone writes `//@ only-x64 only-arm only-wasm`, then they follow this help, but the help contains `//@ only-arm only-wasm` (which is not correct if we reject trailing directives again).

Reply:
Except that I specifically made sure that things like that `//@ only-x64 only-arm only-wasm` would only check `only-arm` and thus the help would only suggest: `//@ only-arm`